### DISCUSSION TITLE
Creating reciprocal linkages to Dryad for genomic data

### DISCUSSION THREAD

**Discussion by campmlc** [ArctosDB/arctos#4969] [General]
I need to create an Other ID for dryad dois. I'd like to confirm I'm doing this correctly, and also see about initiating a discussion with them regarding reciprocal links? 
e.g.
https://datadryad.org/stash/our_platform
https://datadryad.org/stash/dataset/doi:10.5061/dryad.26j38
 We need this quickly for demo of Arctos ability to link to genomic data.

**Comment by dustymc** [comment]
This seems like simple denormalization on the surface

1. The publication already links to the DOI (which currently resolves to dryad).
2. Dryad links back to the publication.
3. Your loan agreement should ensure the publication links to specimens (and projects can act as a fallback).

Everyone can already get everywhere from anything; doing more cannot add functionality, but can exponentially increase the maintenance workload.

What are you trying to do?

**Comment by campmlc** [comment]
I need to insert the DOI into the Arctos record as a link.  Do we have a protocol for linking to DOIs?

**Comment by campmlc** [comment]
The DOI is the link to the actual genomic data. On the surface, this publication does not reference specimens. You have to dig to find out there is genomic data from an Arctos specimen embedded. We need a direct link from the specimen record to the dryad doi containing the genomic data.

**Comment by dustymc** [comment]
>On the surface

That's a problem for the publication, right? From Arctos, someone's already done the digging and created citations (or project, if it's really underground) and the links are front-and-center obvious.

In any case that sounds like a job best suited to Media - if you use the DOI (https://doi.org/10.5061/dryad.26j38) as the media_uri it should be fairly stable.

If you can somehow justify otherIDs - and I don't think you can here without losing functionality (see https://github.com/ArctosDB/arctos/issues/2278) - you could make a new type``DOI`` with a base URL of ``https://doi.org/`` and enter ``10.5061/dryad.26j38`` with the specimen, which would link to https://doi.org/10.5061/dryad.26j38.

**Comment by campmlc** [comment]
Why not do both? Media and Other ID? I think we should have a similar
linkage to genomic data as to GenBank data and IsoBank data.

**Comment by dustymc** [comment]
>both

Well it's all three in this case, and the answer as always is "denormalization bad." If you do this through the publication then everything works because it's part of a system meant to do just this. If something changes you (or others who use this system, and there are lots of them and they have extensive resources) then you fix one thing and it all works again.

If you also do it through otherIDs then you get to choose between maintaining more data (maybe lots more - 43,546 specimens in the case of http://arctos.database.museum/publication/10007289) or having inconsistent data (if you do this for some publication, or some specimens in some publication, then users don't have a single clear path that'll do the same thing for all specimens). You'll get no help maintaining those links; they're just from your specimens.

Add in Media and it's just a doubling; you now have 87092 (plus one, the publication) THINGS to maintain for that one publication, and doing so still doesn't expose any new information.

**Comment by campmlc** [comment]
So if I understand correctly, you are saying I should link the publication
only and that will be the only way this information is discoverable. That
means we would have no explicit way of searching for specimens that have
genomic data. Is there some other way we could flag specimens that have
published genome data to make them discoverable? Currently, we find
specimens with sequence data by looking for specimens that have GenBank
identifiers. We will soon be able to find specimens with isotopic data by
searching on IsoBank identifiers. How do we find specimens with genomic
data, if the data is embedded in a dryad doi?

**Comment by dustymc** [comment]
What's the distinction between "genomic" and "sequence"?

Will these specimens not also have GenBank IDs?

I don't think there's anything in DOI or dryad that would make me think "genomic."

**Comment by campmlc** [comment]
No, they will not have GenBank IDs. The data have been deposited in Dryad.
There is no way to know from the publication link that there is associated
genomic data, but there is. That is why we need to include an identfier to
the dataset, and also why we need some way to flag for the existence of the
genomic data. Solution?

**Comment by campmlc** [comment]
Perhaps we need an Other ID of "Genomic Data" with a value of the url to
whatever repository is used?

**Comment by dustymc** [comment]
>no way to know from the publication link 

I can look harder later, crossref might be tracking supplements (although it's not in a quick check).

> flag for the existence of the genomic data

That's back to the normalization thing - someone will use the flag to find these two records, figure that's all that's in Arctos, and leave. The "solution" is to maintain the flag for everything with "genomics data" which I'd guess is a bit more than double the workload?

>Other ID of "Genomic Data" with a value of the url

That's functionally identical to Media (including the maintenance implications). Looks a lot like more arbitrariness - which of the various methods of doing the same thing did this particular specimen use? - which as always leads to users not getting what they're looking for.

>Solution?

Probably not without some more background information; this still doesn't much make sense to me. Someone didn't cite specimens and did some genomics-thing that didn't result in anything in GenBank, and we need to do something special to capture that?? I'm sure I'm missing something, but I don't know what it could be.

Given what I know now, Media seems the least-evil way to doing this. There's some label ("subject"?) used for similar things; a value of 'genomics data' there might even work well enough for the flag.

**Comment by KyndallH** [comment]
I see both points. It is nice to have the data right on Arctos but there is also the work of putting it on there. I know there are a lot more pubs on Arctos that use dryad - I don't plan to go back and add them to other identifiers if that is the route things go. The DOI is not really an other identifier so it doesn't really fit in there like a GenBank or BoLd number does. I'm not even really up for adding the DOI as an independent publication - again, it comes down to time. We barely keep up with adding publications as it is and if I had to check all for dryad and add that information too. Ugh. I think it really comes down to who needs the data. Researchers can go figure it out themselves. And as of right now, we don't need to know exactly which specimens have genomic data as we are simply counting publications.

**Comment by dustymc** [comment]
I dug around a bit more, the crossref metadata does not contain the supplemental data so that path won't (yet) work. I'm not sure if they don't do that yet (seems unlikely) or everything I could find just didn't supply it (frustratingly common; see ORCID).

I'm not sure there's a "both" points of view here. If ya'll want to include or link or data or whatever, then so do I! My only concern is HOW. Are we painting ourselves into a corner from which future development will have a hard time escaping, misleading users, making a bunch of work for ourselves in maintaining this or cleaning up the giant mess, building something that's going to add the most bang for the least buck, ????

Without a fairly detailed understanding of what we're trying to do - and I do not have that here - I don't really know the answer to those questions. I'm just trying to find the best way to model whatever ya'll want to do.

I see @campmlc added a new otherID with a baseURL. Shall we close this?

**Comment by campmlc** [comment]
I've been in discussion with Joe Cook who has reached out to various
researchers publishing genomic data. The agreement is that we should not
just put out the publication doi and force people to search through to find
genomic data, but there is not yet a standard repository or standard way of
citing. They are working on this. In the meantime, I'm thinking we could
have an ID of "Genomic Data" and link to the url.

**Comment by campmlc** [comment]
--------- Forwarded message ---------
From: Joseph Cook <tucojoe@gmail.com>
Date: Tue, Oct 1, 2019 at 8:30 AM
Subject: Re: [ArctosDB/arctos] Creating reciprocal linkages to Dryad for
genomic data (#2281)
To: Jocelyn Colella <jcolella.jc@gmail.com>


Yes, those are links that should be built (eventually) into Arctos---not
searching through pubs for links.........
You are right, but we need to get some of the faculty players behind this
effort (Mike, Chris, MVZ folks).

On Tue, Oct 1, 2019 at 8:26 AM Jocelyn Colella <jcolella.jc@gmail.com>
wrote:

> SRA definitely. Possibly also link to ENSEMBL? http://ensemblgenomes.org
> --
> Postdoctoral Research Faculty, University of New Hampshire
> MacManes Lab - Evolutionary & Ecophysiological Genomics
> Molecular, Cellular, & Biomedical Sciences Dept.
> 512-567-9843 • jcolella [at] gmail [dot] com • [http://] jpcolella [dot]
> weebly [dot] com
>
>

**Comment by campmlc** [comment]
On Mon, Sep 30, 2019 at 11:22 PM Julia Allen <jallen23@unr.edu> wrote:

> Hey Joe at al.,
>
> Right now GenBank has a specific repository for genomic data, for both raw
> sequencing reads and for genome assemblies. It is called the Sequence Read
> Archive (SRA <https://www.ncbi.nlm.nih.gov/sra>).  As you upload the data
> you can add in any specimen info of your choice.  It is a little more
> tricky trying to search the SRA fir specimen info, but you can search by
> taxon and then look at the data associated with the uploads to see if there
> is specimen data.
>
> The SRA is growing at a rapid pace (and is huge-- as you might imagine).
>
> Hope that was helpful! Feel free to email anytime!
>
> Julie
>
>
> On Mon, Sep 30, 2019 at 7:16 PM Joseph Cook <tucojoe@gmail.com> wrote:
>
>>
>> Hi Folks,
>>
>> Could you weigh in on genome data and Arctos--trying to figure out how to
>> link (like w enow link to GenBank, but GenBank won't be the repository for
>> genomes, correct?)
>>
>> Where will genomes be deposited eventually? (Not Dryad, right?)
>>
>> thanks,
>> Joe
>>
>>
>> ---------- Forwarded message ---------
>> From: Mariel Campbell <notifications@github.com>
>> Date: Mon, Sep 30, 2019 at 12:14 PM
>> Subject: Re: [ArctosDB/arctos] Creating reciprocal linkages to Dryad for
>> genomic data (#2281)
>> To: ArctosDB/arctos <arctos@noreply.github.com>
>> Cc: Subscribed <subscribed@noreply.github.com>
>> > You are receiving this because you were assigned.
>> > Reply to this email directly, view it on GitHub
>> > <
>> https://github.com/ArctosDB/arctos/issues/2281?email_source=notifications&email_token=ADQ7JBEE3GZPABDPVPDHYWLQMIVEZA5CNFSM4I34WI52YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOD76KHQI#issuecomment-536650689
>> >,
>> > or mute the thread
>> > <
>> https://github.com/notifications/unsubscribe-auth/ADQ7JBC5JBK7OJK2LPML5TLQMIVEZANCNFSM4I34WI5Q
>> >
>> > .
>> >
>>
>> —
>> You are receiving this because you are subscribed to this thread.
>>
>

**Comment by dustymc** [comment]
> ID of "Genomic Data" and link to the url

See above; this method still looks like a lot of work for low-quality data to me.

> links that should be built (eventually) into Arctos

Ideally (if I'm at least understanding the basics) this would be be bigger than Arctos. If we can get crossref to supply the DOI of the supplemental data, then implementing and maintaining this should be trivial in Arctos. If we do the same thing we've done for other such data with the mechanism built for that sort of thing (media), then this should at least be predictable (eg, the URL is typed as a URL instead of a string) and manageable.

>SRA

That's been in Arctos for quite some time: http://arctos.database.museum/info/ctDocumentation.cfm?table=CTCOLL_OTHER_ID_TYPE&field=NCBI%20Sequence%20Read%20Archive%20Run%20ID

**Comment by Jegelewicz** [comment]
I am reopening this because I don't believe this is an other identifier for any specific record.

**Comment by Jegelewicz** [comment]
There is only one record using this identifier - https://arctos.database.museum/guid/MSB:Para:20350

And it links to a publication - https://datadryad.org/stash/dataset/doi:10.5061/dryad.26j38

And I don't see anywhere that this specific specimen is cited or that the publication contains any specific genetic information.

**Comment by dustymc** [comment]
>And it links to a publication

I'm still baffled and this still looks like a job for Media, but that is definitely not a publication. The publication is http://dx.doi.org/10.1093/sysbio/syw105, the data used in the publication are https://dx.doi.org/10.5061/dryad.26j38. (They could have also been stashed in dropbox, github, or anything else - dryad is just a sciencey hard drive.)

Note that we've also explicitly tossed about the entire functionality of the DOI ecosystem by making the link to https://datadryad.org/stash/dataset/doi:10.5061/dryad.26j38 for some reason.

<img width="1591" alt="Screen Shot 2022-03-17 at 2 43 32 PM" src="https://user-images.githubusercontent.com/5720791/158899848-bc25d99e-f688-42c0-92e5-133da78060fb.png">
